sql/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -280,21 +276,16 @@ public void bind(String portalName,
             throw t;
         }
 
-        currentTxnCtx = new CoordinatorTxnCtx(sessionContext);
-
-        var unboundStatement = preparedStmt.unboundStatement();
-        final AnalyzedStatement maybeBoundStatement;
-        if (unboundStatement == null || !unboundStatement.isUnboundPlanningSupported()) {
-            ParameterContext parameterContext = new ParameterContext(new RowN(params.toArray()), List.of());
-            Analysis analysis = analyzer.boundAnalyze(
-                preparedStmt.parsedStatement(),
-                currentTxnCtx,
-                parameterContext);
-            maybeBoundStatement = analysis.analyzedStatement();
-        } else {
-            maybeBoundStatement = unboundStatement;
-        }
-        Portal portal = new Portal(portalName, preparedStmt, params, maybeBoundStatement, resultFormatCodes);
+        AnalyzedStatement analyzedStatement = analyzer.unboundAnalyze(
```

Comment:
Done. Regarding the assertion of the equality of analyzed statements, it would require a bit of work such as now we do not implement `equals`/`hashCode` for any of the statements. Do you think we should still proceed with adding the assertion?